p4-16/spec/P4-16-spec.mdk

```diff
@@ -2208,16 +2208,21 @@ structField
 ~ End P4Grammar
 
 where each `typeRef` is restricted to a bit-string type (fixed or
-variable) or an integer type. This declaration introduces a new
-identifier in the current scope; the type can be referred to using
-this identifier. A header is similar to a `struct` in C,
+variable), a fixed-width signed integer type, a boolean type, or a struct that itself contains other 
+struct fields, nested arbitrarily, as long as all of the "leaf" types are
+are `bit<W>`, `int<W>`, a Serializable Enum, or a `bool`.  If a `bool` is used
```

Comment:
Minor recommendation: Replace "Serializable Enum" with "serializable `enum`".  No reason to capitalize it that I can see.